src/olympia/addons/utils.py

```diff
@@ -110,6 +111,8 @@ def get_addon_recommendations(guid_param, taar_enable):
                 if guids == []
                 else TAAR_LITE_FALLBACK_REASON_TIMEOUT
             )
+            taar_lite_outcome = fail_reason
```

Comment:
Maybe use the occasion to rewrite the conditions above into something a bit easier to follow? There are a lot of tests around `guids`... Maybe this:
```python
def get_addon_recommendations(guid_param, taar_enable):
    guids = None
    fail_reason = None
    if taar_enable:
        guids = call_recommendation_server(
            settings.TAAR_LITE_RECOMMENDATION_ENGINE_URL, guid_param, {}
        )
        if guids:
            outcome = TAAR_LITE_OUTCOME_REAL_SUCCESS
            taar_lite_outcome = 'success'
        else:
            outcome = TAAR_LITE_OUTCOME_REAL_FAIL
            fail_reason = (
                TAAR_LITE_FALLBACK_REASON_EMPTY
                if guids == []
                else TAAR_LITE_FALLBACK_REASON_TIMEOUT
            )
            taar_lite_outcome = fail_reason
    else:
        outcome = TAAR_LITE_OUTCOME_CURATED
    if not guids:
        guids = TAAR_LITE_FALLBACKS
    return guids, outcome, fail_reason
```

Reply:
That is a good suggestion, thanks. I have updated the patch.